### ISSUE TITLE
Plugin request: nello one

### ISSUE THREAD

**Issue by gerritfries** [nfarina/homebridge#1718]
Hi there,

I was wondering if somebody is already working on a plugin for "nello one" (https://www.nello.io/en/) or willing to do so in the future as soon as they publish their Public API. If there are any workaround to integrate nello one into Homebridge please let me know.

Thank you,
Gerrit

**Comment by DJay79** [comment]
I would also love to see that. I found a solution in the [fhem forum](https://forum.fhem.de/index.php/topic,75127.msg668871.html?PHPSESSID=q146k5k9of18k1hmk4k32a3cf6#msg668871). I don't know anything about that, but somehow something is possible. It only takes one clever person who packs it up to Homebridge as a plugin. ;)

**Comment by shexbeer** [comment]
We just need to translate this FHEM module to homebridge plugin format:

https://github.com/mhop/fhem-mirror/blob/master/fhem/FHEM/31_Nello.pm

**Comment by Nastras** [comment]
https://github.com/shexbeer/homebridge-nelloio

Here maybe arises a plugin for nello. So far, however, there is no content.

**Comment by shexbeer** [comment]
Short update: Iam able to open the door via HTTP Requests, currently investigating homebridge plugin format.
My current goal is to implement a stateless switch which will open the door. Notifications are bit more complicated

**Comment by ebaauw** [comment]
It looks like a very cool device, but from the documentation is seems to be cloud based.  I really want my home automation to run at my home, without Internet connection.

> My current goal is to implement a stateless switch which will open the door.

Note that the HomeKit stateless programmable switch only works as a representation of a physical switch - you cannot control it from HomeKit.  You need to expose a regular switch and reset it to false automatically after 1 sec or so.

> Notifications are bit more complicated

You should get native HomeKit notifications if you the right Service/Characteristic combi.  I receive notifications for Motion Sensors, Door Sensors, and even Leak Sensors.  There's a Doorbell service, but it only seems to work in a Video Doorbell accessory.

**Comment by shexbeer** [comment]
@ebaauw 
A smart home that runs without internet connection seams like a bit odd to me.

Yes indeed, stateless in sense of a switch witch puts its back to off after a second.
Currently first version of plugin runs successful in my network.

Notification in nello domain means MQTT packet which must be fetched. But nello API is currently not ready to make this in a easy way, so a little bit of a hack i would call is needed to make all work nicely. When MQTT packet is fetched, it is possible to activate a Fake Motionsensor and to enable further automation (or send notfications in iOS domain means)

**Comment by ebaauw** [comment]
> A smart home that runs without internet connection seams like a bit odd to me.

With the current DDOS attacks going on here in the Netherlands, it's only a matter of time before a smart home that depends on an Internet connection won't run at all...

**Comment by shexbeer** [comment]
NPM package will be available tomorrow evening as i got a little error while publishing the module (now have to wait 24 hours to publish again)

If someone is interested now, just clone project and use 'npm link'

**Comment by flxmllr** [comment]
Hi there,
I have got 2 issues when trying to get it up and running:

1)

_[2018-02-19 15:17:50] [Haustür] Send login to Nello.io 'https://api.nello.io' on '/login'
/usr/local/lib/node_modules/homebridge-nelloio/index.js:80
			this._switchService.getCharacteristic(Characteristic.On).updateValue(true);
			                                                         ^

TypeError: this._switchService.getCharacteristic(...).updateValue is not a function_

I fixed it by replacing "updateValue" with "setValue"

2) Then I get an endless loop, here is the log:

_[2018-02-19 15:29:15] [Haustür] setPowerState( 1 )
[2018-02-19 15:29:15] [Haustür] Opening front door
[2018-02-19 15:29:15] [Haustür] Looking for credentials sessionId:'false' locationId:'false' userId:'false'.
[2018-02-19 15:29:15] [Haustür] No session or location id found, will try to login
[2018-02-19 15:29:15] [Haustür] Trying to login
[2018-02-19 15:29:15] [Haustür] Send login to Nello.io 'https://api.nello.io' on '/login'
[2018-02-19 15:29:15] [Haustür] setPowerState( true )
[2018-02-19 15:29:15] [Haustür] Opening front door
[2018-02-19 15:29:15] [Haustür] Looking for credentials sessionId:'false' locationId:'false' userId:'false'.
[2018-02-19 15:29:15] [Haustür] No session or location id found, will try to login
[2018-02-19 15:29:15] [Haustür] Trying to login
[2018-02-19 15:29:15] [Haustür] Send login to Nello.io 'https://api.nello.io' on '/login'
[2018-02-19 15:29:15] [Haustür] setPowerState( true )
[2018-02-19 15:29:15] [Haustür] Opening front door
[2018-02-19 15:29:15] [Haustür] Looking for credentials sessionId:'false' locationId:'false' userId:'false'.
[2018-02-19 15:29:15] [Haustür] No session or location id found, will try to login
[2018-02-19 15:29:15] [Haustür] Trying to login
[2018-02-19 15:29:15] [Haustür] Send login to Nello.io 'https://api.nello.io' on '/login'
[2018-02-19 15:29:15] [Haustür] setPowerState( true )_

... some debugging showed, that the request call "_request(options, function (error, response, body) {_..." seems not being triggered.

Is there anything I can do?

Thank you very much!

Cheers,
Felix

**Comment by lukasroegner** [comment]
Hi! I also wrote a plugin for the nello intercom, as I didn't want to wait for @shexbeer to update his plugin to expose the lock via `LockMechanism` ;)

It works with multiple locks as the accessories are being created dynamically, the timeout before the lock is displayed as locked after you unlock the door can also be configured. Feel free to request features.

I also walked through the FHEM plugin, however I'm not an expert with Perl...I want to extend the plugin to also expose the `Doorbell` service, so that users are notified when someone rang the bell.
